### PR TITLE
Some more Chaplain QoL

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Allows for the Altar of the Gods to also function on any tile that was blessed with Holy Water.
Adds a tooltip showing that you can [un]wrench the altar by using your holy weapon on it.

## Why It's Good For The Game

Gives Chaplain players a bit more freedom on where they conduct their rutials. Also it was asked for on the forums - https://forums.beestation13.com/t/alter-of-the-gods-pilgrimagless/22637
The tooltip makes things a bit more intuitive.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Tooltip : 
https://user-images.githubusercontent.com/110184118/236636758-fb39a818-ca33-414c-b611-35bcb910c28a.mp4

Using the altar on a blessed tile :
https://user-images.githubusercontent.com/110184118/236636760-40b838c5-696e-4169-98e0-ad97ec0019dd.mp4

</details>

## Changelog
:cl:
add: Added a tooltip to the Altar of the Gods showing that you can un/wrench it with the Null Rod
tweak: allowed for the Altar of the Gods to be used on any tile blessed by Holy Water
code: Added a turf proc is_holy(), which checks if said turf is in the chapel area or has the holy effect applied to it
code: Replaced checks for the holy effects with is_holy checks
tweak: Made all holy roles be able to see blessed tiles rather than just the chaplain (This means Holy Undead will be able to see them too)
/:cl:

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -733,3 +733,11 @@ GLOBAL_LIST_EMPTY(created_baseturf_lists)
...
 			continue
 		. += turf_to_check
+
+/// Checks if the turf was blessed with holy water OR the area its in is Chapel
+/turf/proc/is_holy()
+	if(locate(/obj/effect/blessing) in src)
+		return TRUE
+	if(istype(loc, /area/chapel))
+		return TRUE
+	return FALSE
```
Comment (PowerfulBacon) [inline code/game/turfs/turf.dm:587]: Should probably replace all instances that check for holy precence with this proc. For example, EMPs and explosions sourced from unholy energies can be blocked by having a blessing on the ground.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/game/turfs/turf.dm (hunk 1 above)
Comment (CydiaLamiales) [reply code/game/turfs/turf.dm:587]: That'll make the chapel immune to said EMPs and explosions immune to it too, is that okay?

Comment (PowerfulBacon) [reply code/game/turfs/turf.dm:587]: Yes, that's fine. They are holy locations

**Review by itsmeow — APPROVED** [review]

File: code/datums/components/religious_tool.dm [hunk 2]
```diff
@@ -242,6 +241,9 @@
 	if(!can_i_see)
 		return
 	examine_list += ("<span class='notice'>Use a bible to interact with this.</span>")
+	if(user.mind?.holy_role != NONE)
```
Comment (PowerfulBacon) [inline code/datums/components/religious_tool.dm]: Is this meant to be none, or a null check? Isn't none for bitflag checks?

Comment (CydiaLamiales) [reply code/datums/components/religious_tool.dm]: image

Comment (CydiaLamiales) [reply code/datums/components/religious_tool.dm]: So yes, it was meant to be NONE because its default value is NONE

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
`user.mind?.holy_role != NONE` is not the same as `user.mind?.holy_role`.
`user.mind?.holy_role != NONE` will give you incorrect behaviours when mind is null, as this will cause the left side to be null and the right side to be NONE (0). You should only use the left side format when performing equality checks, if you don't care what role it is, just that they have any role, then use `user.mind?.holy_role`.

image

File: code/datums/components/religious_tool.dm (hunk 2 above)
Comment (CydiaLamiales) [inline code/datums/components/religious_tool.dm]:
```suggestion
	if(user.mind?.holy_role)
```

File: code/game/alternate_appearance.dm [hunk 3]
```diff
@@ -170,7 +170,7 @@ GLOBAL_LIST_EMPTY(active_alternate_appearances)
 			add_hud_to(mob)
 
 /datum/atom_hud/alternate_appearance/basic/blessedAware/mobShouldSee(mob/M)
-	if(M.mind && (M.mind.assigned_role == JOB_NAME_CHAPLAIN))
+	if(M.mind && M.mind?.holy_role != NONE)
```
Comment (CydiaLamiales) [inline code/game/alternate_appearance.dm]:
```suggestion
	if(M.mind && M.mind?.holy_role)
```

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**